crates/meroctl/src/cli/context/alias.rs

```diff
@@ -64,3 +77,33 @@ impl ContextAliasCommand {
         Ok(())
     }
 }
+
+async fn context_exists(
+    multiaddr: &Multiaddr,
+    identity: &Keypair,
+    target_id: &ContextId,
+) -> EyreResult<bool> {
+    let url = multiaddr_to_url(multiaddr, &format!("admin-api/dev/contexts/{}", target_id))?;
+
+    let result = do_request::<_, GetContextResponse>(
+        &Client::new(),
+        url,
+        None::<()>,
+        identity,
+        RequestType::Get,
+    )
+    .await;
+
+    match result {
+        Ok(_) => Ok(true),
+        Err(err) => {
+            // Check if the error is a 404 Not Found
+            if let Some(status) = err.downcast_ref::<reqwest::StatusCode>() {
```

Comment:
Shouldn't you be downcasting to `ApiError` instead?

https://github.com/calimero-network/core/blob/d513acff5d67850ddf581b511bb6f3074153ec75/crates/meroctl/src/common.rs#L77-L80